scaledimage.cpp

```diff
@@ -84,7 +90,6 @@ void ScaledImage::scale(double factor, double angle) {
 	}
 	
 	this->scaledImage=sdlCreateSurface((int)(this->image->w*65536.0*factor)>>16, (int)(this->image->h*65536.0*factor)>>16,1);
-	this->isScaled=1;

```

Comment:
Are you sure this change actually does what you expect? To me it seems isScaled is just a flag signalling usage of scaledImage member. I have not delved much into the code, though. Just please double check you know what you are doing.

Reply:
sure I try to do a test run with every change. See `blit` member, if we dont remove that statement u will always have "imageScaled" blitted to Surface

Reply:
For me it looks like a memory leak scenario. If you call `ScaledImage::scale` twice without `isScaled` flag set, you end up with an unallocated `scaledImage` surface. I would rather not call `scale` at all, if I was not interested in a scaled image. `scale` could possibly be called dependent on `isScaled` flag, but set in `init` method - it is just an idea which also needs to be reviewed before actual implementation.

Run valgrind to be 100% sure.

Reply:
The flag is set in `init` member and if that's true then `isScaled` was badly written in the first place, which has been already proven. However not planning to do a bigger redraw ATM.

Probably some return func would suffice to not call scale twice, but haven't figure out it.

Reply:
Unfortunately, it is your changes that could be problematic in terms of memory management here, so I cannot merge them until the issue is solved.